src/mon/OSDMonitor.cc

```diff
@@ -606,6 +607,248 @@ int OSDMonitor::reweight_by_utilization(int oload, std::string& out_str,
   return changed;
 }
 
+inline string stringify2(const double& a) {
+  stringstream ss;
+  if (a < -0.01)
+    ss << "-";
+  else if (a < 0.01)
+    ss << "0";
+  else
+    ss << std::fixed << std::setprecision(2) << a;
+  return ss.str();
+}
+
+void OSDMonitor::print_utilization(ostream *out, Formatter *f) const
+{
+  if (f)
+  {
+    return; // XXXMG: Not implemented yet.
+  }
+
+  TextTable tbl;
+
+  if (out) {
+    tbl.define_column("ID", TextTable::LEFT, TextTable::LEFT);
+    tbl.define_column("WEIGHT", TextTable::LEFT, TextTable::RIGHT);
+    tbl.define_column("REWEIGHT", TextTable::LEFT, TextTable::RIGHT);
+    tbl.define_column("%UTIL", TextTable::LEFT, TextTable::RIGHT);
+    tbl.define_column("VAR", TextTable::LEFT, TextTable::RIGHT);
+  }
+
+  const PGMap &pgm = mon->pgmon()->pg_map;
+
+  double average_util = (double)pgm.osd_sum.kb_used / (double)pgm.osd_sum.kb;
+  double min_var = -1;
+  double max_var = -1;
+  double stddev = 0;
+
+  for (ceph::unordered_map<int,osd_stat_t>::const_iterator p =
+	 pgm.osd_stat.begin();
+       p != pgm.osd_stat.end();
+       ++p) {
+    int id = p->first;
+    double weight = osdmap.crush->get_item_weightf(id);
+    double reweight = osdmap.get_weightf(id);
+    double util = 100 * (double)p->second.kb_used / (double)p->second.kb;
+    double var = util / average_util;
+
+    if (min_var < 0 || var < min_var) min_var = var;
+    if (max_var < 0 || var > max_var) max_var = var;
+
+    double k = weight > 0.01 ? reweight / weight : 1;
```

Comment:
Also, I don' think we should have a .01 threshold in here.. weights of .0001 or whatever should be just fine.  If the weight is 0, we should exclude it from the calculation entirely.  I think that means you need to count 'n' instead of using pgm.osd_stat.size() in the final stddev calc.